src/GitHub.App/ViewModels/PullRequestDetailViewModel.cs

```diff
@@ -1,23 +1,508 @@
+        /// is <see cref="CheckoutMode.NeedsPull"/>.
+        /// </summary>
+        public int CommitsBehind
+        {
+            get { return commitsBehind; }
+            private set { this.RaiseAndSetIfChanged(ref commitsBehind, value); }
+        }
+
+        /// <summary>
+        /// Gets a message indicating the why the <see cref="Checkout"/> command is disabled.
+        /// </summary>
+        public string CheckoutDisabledMessage
+        {
+            get { return checkoutDisabledMessage; }
+            private set { this.RaiseAndSetIfChanged(ref checkoutDisabledMessage, value); }
+        }
+
+        /// <summary>
+        /// Gets the changed files as a tree.
+        /// </summary>
+        public IReactiveList<IPullRequestChangeNode> ChangedFilesTree { get; } = new ReactiveList<IPullRequestChangeNode>();
+
+        /// <summary>
+        /// Gets the changed files as a flat list.
+        /// </summary>
+        public IReactiveList<IPullRequestFileViewModel> ChangedFilesList { get; } = new ReactiveList<IPullRequestFileViewModel>();
+
+        /// <summary>
+        /// Gets a command that checks out the pull request locally.
+        /// </summary>
+        public ReactiveCommand<Unit> Checkout { get; }
+
+        /// <summary>
+        /// Gets a command that opens the pull request on GitHub.
+        /// </summary>
+        public ReactiveCommand<object> OpenOnGitHub { get; }
+
+        /// <summary>
+        /// Gets a command that toggles the <see cref="ChangedFilesView"/> property.
+        /// </summary>
+        public ReactiveCommand<object> ToggleChangedFilesView { get; }
+
+        /// <summary>
+        /// Gets a command that toggles the <see cref="OpenChangedFileAction"/> property.
+        /// </summary>
+        public ReactiveCommand<object> ToggleOpenChangedFileAction { get; }
+
+        /// <summary>
+        /// Gets a command that opens a <see cref="IPullRequestFileViewModel"/>.
+        /// </summary>
+        public ReactiveCommand<object> OpenFile { get; }
+ 
+        /// <summary>
+        /// Gets a command that diffs a <see cref="IPullRequestFileViewModel"/>.
+        /// </summary>
+        public ReactiveCommand<object> DiffFile { get; }
+
+        /// <summary>
+        /// Initializes the view model with new data.
+        /// </summary>
+        /// <param name="data"></param>
         public override void Initialize([AllowNull] ViewWithData data)
         {
-            System.Windows.MessageBox.Show(String.Format(CultureInfo.InvariantCulture, "{0}", data.Data));
+            var prNumber = (int)data.Data;
+
+            IsBusy = true;
+
+            // TODO: Catch errors.
+            Observable.CombineLatest(
+                    repositoryHost.ApiClient.GetPullRequest(repository.Owner, repository.CloneUrl.RepositoryName, prNumber),
+                    repositoryHost.ApiClient.GetPullRequestFiles(repository.Owner, repository.CloneUrl.RepositoryName, prNumber).ToList(),
```

Comment:
Yes, I did originally start writing it using `ModelService` but then I remembered how we ended up essentially bypassing the caching for the PR list, so I went straight to the API. The question I wanted to discuss with you was do our reasons for bypassing the cache for the PR list apply here?